### ISSUE TITLE
Translator's context for 5.0

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#2684]
Checklist (as suggested by @timbze ) from last release

- Czech @davidhrbac
- German
- Russian 
- Finnish @tuomas2 
- French
- Hungarian @krisek
- Portuguese
- Portuguese (Brazilian) @pepaulo 
- Afrikaans
- India (Hindi) @kesarsauce 
- South Korea 
- Lithuanian
- Spanish @timbze 
- Polish
- Arabic
- Italian @LAfricain
- Slovenian
- Chinese (CN&TW)
- Ukrainian
- Serbian (latin & rs)
- Kazakh
- Esperanto @verdulo 
- Norweigian
- Turkish @alerque 
- Estonian
- Azerbaijani 
- Dutch @marnix
- Slovak
- Uzbek
- Kyrgyz

[Excel version in Google Sheets](
https://docs.google.com/spreadsheets/d/1NHe7YtnRQ1lwEXh7Y3hf3V7CmsUirr6wIvsUX-sADeA/edit#gid=0)

Related: 
- #1028

**Comment by tuomas2** [comment]
Sent announcement to translators via Transifex announcements

> New AndBible release 5.0 is planned within the coming few months, during the autumn most likely. There are some major new features in the new release:
> 
> - Cloud synchronize (via Google drive currently)
> - Discrete mode in main app and separate “calculator” app
> - Workspace colors
> - Chained links windows
> - Synchronize groups
> - Search / filter labels
> - Backup Database: choose what to restore
> - MyBible / MySword modules,  Installing, uninstalling, Backup & restore
> - Custom repositories
> - Deep links 
> 
> Some of these features require new strings to be translated. Amount of new strings is, however, not very big, like last time. 
> 
> Before starting translating, please see video intro to new features; I believe that it serves best as context for this release: https://www.youtube.com/watch?v=XdKmoe95KhI. See also translator's context ticket, in which you can also send questions and comments if they arise: https://github.com/AndBible/and-bible/issues/2684 
> 
> As always, translator's instructions can be read from our wiki page: https://github.com/AndBible/and-bible/wiki/Translating-User-Interface
> 
> As we are in Beta phase, we are mainly fixing bugs and smoothing the rough edges. Although there can be still minor changes to strings, it is good time to start updating translations.
> 
> Thank you for your continued support and co-operation for the AndBible Open Source project!
> 
> Best regards & God bless,
> Tuomas
> Lead developer

**Comment by tuomas2** [comment]
Added new instructions about translating BibleNames to https://github.com/AndBible/and-bible/wiki/Translating-User-Interface. Translating BibleNames properly is now more important than earlier because bible reference detection is used heavily in Epub support.

**Comment by tuomas2** [comment]
BibleNames translation process needs improvement. Please do not work on those yet. Fixing today or tomorrow.

**Comment by tuomas2** [comment]
> BibleNames translation process needs improvement. Please do not work on those yet. Fixing today or tomorrow.

Now it's fixed. I'll write instructions tomorrow.

**Comment by tuomas2** [comment]
Sending this via Transifex announcement soon.

****

Final call for translators for upcoming AndBible 5.0 release (release scheduled 2nd nov 2023)

New AndBible release 5.0 is almost ready and beta is already rather stable. Release is scheduled to happen November 2nd, 2023, in 2 months time from now. Translation work of user interface translations to various languages should be started as soon as possible. I’m not any more adding new strings or changing existing ones unless it is very critical for some reason. 

I was just doing myself translations to Finnish language and it took me around an hour or so to finish them. I was also adding contextual help (“Developer notes”) to some strings, so hopefully context is clear and translation work can be done smoothly. Please do not hesitate to ask me if something is unclear! Preferrably via Transifex issues or via email: [help.andbible@gmail.com](mailto:help.andbible@gmail.com) or to Github translators context ticket (link below).

As always, this message (and possible corrections/minor additions) will be posted on Translator’ context ticket: https://github.com/AndBible/and-bible/issues/2684 so please keep checking that before you start the work.

Couple of important notes for the translation work.

- Why it is now important to translate also “Alt” entries in BibleNames resource (because of Epub support) and how to do it: https://github.com/AndBible/and-bible/wiki/Translating-User-Interface#biblenames-alt-names  
- Argument numbering is now added to those strings that have more than one argument.  This is necessary and has been requested because some languages might translate the content in different order than original. If there is no change to the translation, those can be fixed simply by using “Copy” button from translation history and just fixing arguments. 

Many thanks to all your translation efforts! God bless!

**Comment by zdposter** [comment]
Will be there pre-release for checking translation quality?

**Comment by tuomas2** [comment]
> Will be there pre-release for checking translation quality?

Yes, I will include translations in beta releases. I try to  remember to pull new additions to every release (releasing beta releases about once per week).

Beta releases can be found via [Github releases page](https://github.com/AndBible/and-bible/releases)

**Comment by zdposter** [comment]
I and not able to modified some translations because there were "reviewed" (even I am author of that translation). Can it be unlocked, so translation could be improved?

**Comment by zdposter** [comment]
My translation (Slovak) is 100% translated, but I see there some part of application with English string. Can somebody have a look at that?

![missing_translation_1](https://github.com/AndBible/and-bible/assets/76629990/0df98f62-c967-46b7-a26e-e7724940d6fd)

![missing_translation_2](https://github.com/AndBible/and-bible/assets/76629990/6c78eca4-ad14-4ea2-83a0-85fa18b03e51)

![missing_translation_create_task](https://github.com/AndBible/and-bible/assets/76629990/6d1d5e31-cc75-42d0-a585-71e3874ebdce)

**Comment by tuomas2** [comment]
> My translation (Slovak) is 100% translated, but I see there some part of application with English string. Can somebody have a look at that?

Dictionary, Translate etc come from apps (like Google Translate) that insert their actions to context menu.

**Comment by tuomas2** [comment]
> I and not able to modified some translations because there were "reviewed" (even I am author of that translation). Can it be unlocked, so translation could be improved?

Made you coordinator, hope it helps

**Comment by pepaulo** [comment]
I have translated to Brazilian Portuguese 100% of strings. But the translation group needs of a revisor.

**Comment by tuomas2** [comment]
Thank you @pepaulo! I'm in the process of releasing beta 5.0.728 which includes your translation (should be out via Google Play in a day or two, apk from Github might be there already now).

**Comment by pepaulo** [comment]
> Thank you @pepaulo! I'm in the process of releasing beta 5.0.728 which includes your translation (should be out via Google Play in a day or two, apk from Github might be there already now).

I thank you @tuomas2  for putting your gifts at the service of this excellent evangelization tool.

**Comment by tuomas2** [comment]
Sending reminder to translators

> **Subject: Reminder: 1 more month to go until AndBible 5.0 release (scheduled on 2nd Nov 2023)**
> Hi,
> 
> Release date is getting closer every day.  Release is scheduled to happen November 2nd, 2023, in 2 months time from now. 
> 
> There are many languages already 100% translated. However, there are substantial amount of languages, that were 100% translated for 4.0, but have not yet been updated for 5.0. If the translation of your language is not yet updated, please update it as soon as possible. Work should not be that laboursome as in the past (4.0 was huge with this respect). 
> 
> As always, this message (and possible corrections/minor additions) will be posted on Translator’ context ticket: https://github.com/AndBible/and-bible/issues/2684 so please keep checking that before you start the work.
> 
> Many thanks to all your translation efforts! God bless!

**Comment by tuomas2** [comment]
Additional note: I just released [5.0.730-beta](https://github.com/AndBible/and-bible/releases/tag/beta-730) that contains all latest translations. Soon in Google Play beta channel too.

**Comment by LAfricain** [comment]
Hi, a question, how to change an verified string? I found string which I can't correct.

**Comment by LAfricain** [comment]
If you can do it quicky, I will check it tomorrow again. I red almost all strings with the app.

**Comment by tuomas2** [comment]
@LAfricain made you coordinator of Italian, you should be able to edit all strings now.

**Comment by tuomas2** [comment]
Posting update announcement:

Subject: **AndBible 5.0 translation status update (4 weeks to go)**

> Hi, 
> 
> Release is scheduled to happen November 2nd, 2023. Codewise everything is ready by now for the release, but, we are waiting especially for those languages to finish translations that were complete in 4.0, i.e.:
> - Chinese
> - Czech
> - Russian
> - Hungarian
> - Esperanto
> - India
> - Lithuanian
> - Polish
> - Portuguese
> - Serbian
> - Spanish
> - Burmese
> 
> I have also added couple of new translators to some languages (there were pending requests). New people please see instructions for translators:
> 
> https://github.com/AndBible/and-bible/wiki/Translating-User-Interface
> 
> As always, this message (and possible corrections/minor additions) will be posted on Translator’ context ticket: https://github.com/AndBible/and-bible/issues/2684 so please keep checking that before you start the work. 
> 
> Tip: I’m posting tips & tricks and other videos (especially after 5.0 release) now somewhat regularly in our Youtube channel: https://www.youtube.com/@AndBible 
> 
> Many thanks to all your translation efforts! God bless!
> 
> T: Tuomas

**Comment by tuomas2** [comment]
Please let me know about your plans:

Esperanto @verdulo
India @kesarsauce
Hungarian @krisek

**Comment by tuomas2** [comment]
**AndBible 5.0 translation status update (3 weeks to go)**

> Hi, 
> 
> Release is scheduled to happen November 2nd, 2023, which is 3 weeks from now. We are still 
> waiting especially for:
> 
> - Korean (!!!)
> - Russian
> - Chinese
> - Hungarian
> - Spanish
> - Polish
> - Lithuanian
> - Portuguese
> - Burmese
> 
> Mostly those languages were translated 100% in 4.0 so there should not be that much work.
> 
> According to Google Play statistics Korean (South Korea) language would be very important for AndBible.
> The biggest individual language in Play Store visitors last month were South Koreans (even bigger than English!)! 
> Please pray we will get Korean translation updated!
> 
> As always, this message (and possible corrections/minor additions) will be posted on Translator’ 
> context ticket: https://github.com/AndBible/and-bible/issues/2684.
> 
> Many thanks to all your translation efforts! God bless!
> 
> T: Tuomas

**Comment by alerque** [comment]
Is the Transifex import a one way street or can we update sources directly? I have a *bunch* of work to do to get the Turkish translation up to speed and would much rather use the tooling in my editor to work on `app/src/main/res/values-tr/strings.xml` for at least some chunk of that work. If I were to do that and send a PR will the Transifex data set get updated?

**Comment by tuomas2** [comment]
@alerque if you are the only translator working on that translation, that could work but if there are multiple people working on that particular translation it might cause mess with merging work. I can push yout translations via tx CLI back to transifex.

**Comment by alerque** [comment]
@tuomas2 Thanks. I don't think anybody else has contributed for a very long time, so I'll probably go ahead and see if I can knock out some work with my own tooling. If merge issues come up please feel free to pass them off to me for resolution: including discarding my edits in favor of not loosing other people's work and making me figure out how to re-apply changes.

**Comment by tuomas2** [comment]
Let me check first if the translations in the repository are up to date (it might not be if the translation percent is too small; then tx pull does not pull translations)....

**Comment by tuomas2** [comment]
> Let me check first if the translations in the repository are up to date (it might not be if the translation percent is too small; then tx pull does not pull translations)....

... done. Now dev should be matching with transifex. Also master branch in [jsword](https://github.com/AndBible/jsword/) is checked to be up to date.

**Comment by tuomas2** [comment]
Weekly reminder going to translators as an announcement.

**Subject: AndBible 5.0 translation status update (2 weeks to go)**

Hi, 

Release is scheduled to happen November 2nd, 2023, which is 2 weeks from now.

Most of the important languages seem to be now more or less done. The following languages that were 100% translated in 4.0 still miss many translations:
- Polish
- Portuguese

As noted last time, Korean is very important one, and would be very nice to have it 100% translated.

The following languages are missing just few translations:
- Chinese
- Dutch
- Hebrew
- Hungarian
- Tamil
- Spanish

Hope I got everything right in the above listings. Many thanks to all your translation efforts! God bless!

T: Tuomas

**Comment by tuomas2** [comment]
@alerque fixed some strings in turkish, please check 06eefec9b661b9efdc81179df70ab26c34b7d8b9

**Comment by tuomas2** [comment]
Weekly reminder going to translators as an announcement.

**Subject: AndBible 5.0 translation status update (1 weeks to go)**

Hi, 

Release is scheduled to happen November 2nd, 2023, which is 1 weeks from now. At the moment I am not seeing any reason to postpone release so I believe it is happening as scheduled.

I'm happy to see many languages finished or on their way towards 100%. 

If you haven't yet updated your language translations yet to AndBible, now it's last moment to do it before the release. 

After release we often get 1-star reviews if translations are not in place, so please consider updating if not already. 

Many thanks to all your translation efforts! God bless!
Tuomas

**Comment by verdulo** [comment]
I am sorry for not responding, I had another problems in my life. Polish and Esperanto translations are now completed (including the resource BibleNames). I hope I will translate the future strings on time.
